### PR TITLE
chore: auditBundleOne fixes [SUP-7613, SUP-7599, SUP-7612, SUP-7609, SUP-7608, SUP-7604, SUP-7603, SUP-7600, SUP-7658, SUP-7605, SUP-7601]

### PR DESCRIPTION
**Pull request by sujithsomraaj**

### CODE REVIEW

File: src/crosschain-data/adapters/hyperlane/HyperlaneImplementation.sol [hunk 1]
```diff
@@ -151,8 +151,11 @@ contract HyperlaneImplementation is IAmbImplementation, IMessageRecipient {
             revert Error.INVALID_CHAIN_ID();
         }
 
+        address authImpl = authorizedImpl[domain];
```
Comment (gpersoon) [inline src/crosschain-data/adapters/hyperlane/HyperlaneImplementation.sol:154]: `WormholeARImplementation.sol` also has `authorizedImpl[]`, maybe add some more checks there too

File: src/forms/ERC5115Form.sol [hunk 2]
```diff
@@ -516,17 +516,12 @@ contract ERC5115Form is IERC5115Form, BaseForm, LiquidityHandler {
...
         uint256 sharesBalanceAfter = v.balanceOf(sharesReceiver);
 
-        if (
-            (sharesBalanceAfter - sharesBalanceBefore != shares)
-                || (
-                    ENTIRE_SLIPPAGE * shares
-                        < singleVaultData_.outputAmount * (ENTIRE_SLIPPAGE - singleVaultData_.maxSlippage)
-                )
-        ) {
+        if ((sharesBalanceAfter - sharesBalanceBefore != shares) || (ENTIRE_SLIPPAGE * shares < outputAmountMin)) {
```
Comment (gpersoon) [inline src/forms/ERC5115Form.sol]: it seems there is factor error of ~ `ENTIRE_SLIPPAGE = 10_000`

previousely there was `ENTIRE_SLIPPAGE` on both sides, now just on one side:
```solidity
ENTIRE_SLIPPAGE * shares
                        < singleVaultData_.outputAmount * (ENTIRE_SLIPPAGE - singleVaultData_.maxSlippage)
```

Comment (gpersoon) [reply src/forms/ERC5115Form.sol]: oh wait, this is changed in a later commit, have to check again

File: src/forms/ERC5115Form.sol [hunk 3]
```diff
@@ -552,18 +560,18 @@ contract ERC5115Form is IERC5115Form, BaseForm, LiquidityHandler {
...
 
         uint256 assetsBalanceAfter = IERC20(vaultTokenOut_).balanceOf(assetsReceiver);
 
-        /// @dev reset allowance to wrapper
-        if (underlyingVault.allowance(address(this), vault) > 0) underlyingVault.forceApprove(vault, 0);
+        if (assets < TOLERANCE_CONSTANT) revert Error.WITHDRAW_ZERO_COLLATERAL();
 
         if (
-            (assetsBalanceAfter - assetsBalanceBefore != assets)
+            (assetsBalanceAfter - assetsBalanceBefore < assets - TOLERANCE_CONSTANT)
```
Comment (gpersoon) [inline src/forms/ERC5115Form.sol:574]: Note: this depends on the number of decimals too, but even in extreme cases if decimals would be 2, this is still 0.1
so I guess neglectable

File: src/forms/ERC5115Form.sol [hunk 4]
```diff
@@ -217,6 +220,7 @@ contract ERC5115Form is IERC5115Form, BaseForm, LiquidityHandler {
         override
         returns (uint256 shares)
     {
+        try this.claimRewardTokens() { } catch { }
```
Comment (gpersoon) [inline src/forms/ERC5115Form.sol:223]: not sure, but maybe this is also relevant for `_xChainDepositIntoVault()` ?

File: src/forms/ERC5115Form.sol [hunk 5]
```diff
@@ -310,12 +316,17 @@ contract ERC5115Form is IERC5115Form, BaseForm, LiquidityHandler {
...
+        uint256 balanceBefore = IERC20(vaultTokenIn).balanceOf(address(this));
         /// @dev pulling from sender, to auto-send tokens back in case of failed deposits / reverts
         IERC20(vaultTokenIn).safeTransferFrom(msg.sender, address(this), singleVaultData_.amount);
 
         /// @dev allowance is modified inside of the IERC20.transferFrom() call
         IERC20(vaultTokenIn).safeIncreaseAllowance(vaultLoc, singleVaultData_.amount);
 
+        /// @dev to account for tokens with rounding issues during transfer like stETH
+        /// @dev please refer: https://github.com/lidofinance/lido-dao/issues/442
+        singleVaultData_.amount = IERC20(vaultTokenIn).balanceOf(address(this)) - balanceBefore;
```
Comment (gpersoon) [inline src/forms/ERC5115Form.sol:328]: for safety maybe check that this deviates at the most TOLERANCE_CONSTANT from the previous value of singleVaultData_.amount

File: src/forms/ERC5115Form.sol [hunk 6]
```diff
@@ -516,17 +527,14 @@ contract ERC5115Form is IERC5115Form, BaseForm, LiquidityHandler {
         address sharesReceiver = singleVaultData_.retain4626 ? singleVaultData_.receiverAddress : address(this);
 
         uint256 sharesBalanceBefore = v.balanceOf(sharesReceiver);
+        uint256 outputAmountMin =
+            (singleVaultData_.outputAmount * (ENTIRE_SLIPPAGE - singleVaultData_.maxSlippage)) / ENTIRE_SLIPPAGE;
+
```
Comment (gpersoon) [inline src/forms/ERC5115Form.sol:532]: This way there are more rounding error that in the previous version.
But this is only relevant for small amounts like `outputAmount == 10` , see example below

``` solidity
// SPDX-License-Identifier: none
pragma solidity ^0.8.26;
import "hardhat/console.sol"; 

contract test   {
    uint256 internal constant ENTIRE_SLIPPAGE = 10_000;
    constructor() {
       uint outputAmount = 10;
       uint maxSlippage = 500;
       uint256 outputAmountMin =  ( outputAmount * (ENTIRE_SLIPPAGE - maxSlippage)) / ENTIRE_SLIPPAGE;
       console.log(outputAmountMin);  // 9 --> which is 10% lower 
    }
}
```

File: src/forms/ERC5115Form.sol [hunk 7]
```diff
@@ -516,17 +527,14 @@ contract ERC5115Form is IERC5115Form, BaseForm, LiquidityHandler {
...
         uint256 sharesBalanceAfter = v.balanceOf(sharesReceiver);
 
-        if (
-            (sharesBalanceAfter - sharesBalanceBefore != shares)
-                || (
-                    ENTIRE_SLIPPAGE * shares
-                        < singleVaultData_.outputAmount * (ENTIRE_SLIPPAGE - singleVaultData_.maxSlippage)
-                )
-        ) {
+        if ((sharesBalanceAfter - sharesBalanceBefore != shares) || shares < outputAmountMin) {
```
Comment (gpersoon) [inline src/forms/ERC5115Form.sol:537]: In another part of the code, a deviation of  TOLERANCE_CONSTANT is allowed. Maybe relevant here too?

File: src/forms/ERC5115Form.sol (hunk 3 above)
Comment (vikramarun) [reply src/forms/ERC5115Form.sol:587]: this is fine as is imo, would be very odd cases (tokens mostly have >6 decimals) where this wouldn't help

**Review by 0xTimepunk — APPROVED** [review]
LGTM

**Pull request merged**